include/libremidi/writer.cpp

```diff
@@ -159,25 +160,25 @@ add_event_track_count_check(std::vector<midi_track>& tracks, int track)
         "Refusing to add an event to track " + std::to_string(track)
         + " ; change add_event_track_count_check in libremidi writer.cpp to increase the limit.");
 
-  while (tracks.size() < track + 1)
-    tracks.push_back({});
+  while (tracks.size() < static_cast<std::size_t>(track + 1)) // NOLINT(*-misplaced-widening-cast)
+    tracks.emplace_back();
 }
 }
 
 LIBREMIDI_INLINE
-void writer::add_event(int tick, int track, message m)
+void writer::add_event(const int tick, const int track, const message& m)
 {
   util::add_event_track_count_check(tracks, track);
 
-  tracks[track].push_back({tick, track, m});
+  tracks[static_cast<std::size_t>(track)].push_back({tick, track, m});
 }
 
 LIBREMIDI_INLINE
-void writer::add_event(int track, track_event m)
+void writer::add_event(int track, const track_event& m)
 {
   util::add_event_track_count_check(tracks, track);
 
-  tracks[track].push_back(m);
+  tracks[static_cast<std::size_t>(track)].push_back(m);
```

Comment:
I'm a bit ambivalent about having to cast to unsigned before indexing into a vector, do you really have a warning for this and do you feel it does prevent bugs?

Reply:
Yeah, that was part of the signed-conversion warnings that was showing up, since vector only provides array access operators that index using size_ so only unsigned indexes can "safely" be promoted up and not cause warnings.
I don't really think it'll prevent any bugs since the conversion will just result in a large number that would index out of bounds anyway which would already be an issue. 
The only real way to prevent bugs for these would be to use the bounds checked ::at() function which would throw an exception if it was an invalid index.

Reply:
I went ahead and left these as is for now. I think the only way to really clean this up would be to try and just make the argument type already unsigned, Especially in this case where tracks are already limited to uint16_t by midi itself.